librosa/util/utils.py

```diff
@@ -2214,7 +2214,7 @@ def dtype_c2r(d, *, default=np.float32):
     mapping = {
         np.dtype(np.complex64): np.float32,
         np.dtype(np.complex128): np.float64,
-        np.dtype(complex): np.dtype(np.float).type,
+        complex: float,
```

Comment:
I believe this should be `np.dtype(complex): np.dtype(float).type`.  That way we always return a `dtype`, and it maintains precision parity with the input complex type.  This is how it was done in `dtype_r2c` above.

Reply:
I've done this, but why do we need to use `np.dtype(float).type` and not just `np.dtype(float)`?

Reply:
For consistency - the first two cases return either `np.float32` or `np.float64`, and the third case is meant to map to one of these two as well.  If we return `np.dtype(float)` we get an output of type:
```python
In [9]: type(np.dtype(float))
Out[9]: <class 'numpy.dtype[float64]'>
```
This is a subclass of `type`, but it's not the same type that we'd get if we return `np.float32` directly:
```python
In [11]: type(np.float32)
Out[11]: <class 'type'>
```
Returning `np.dtype(float).type` gives
```python
In [10]: type(np.dtype(float).type)
Out[10]: <class 'type'>
```

Reply:
Thank you for your clarification.
I didn't notice I didn't push my change. Is there anything else needed?

Reply:
Nope, LGTM, thanks!  I'll let the CI run and if it's all clear, we can merge.